src/main/java/org/checkerframework/checker/noliteral/NoLiteralVisitor.java

```diff
@@ -0,0 +1,230 @@
+      final AnnotatedTypeMirror methodDefinitionReceiver,
+      final AnnotatedTypeMirror methodCallReceiver) {
+
+    return true;
+  }
+
+  @Override
+  protected void checkConstructorResult(
+      AnnotatedExecutableType constructorType, ExecutableElement constructorElement) {
+    // Skip this and do nothing. Normally, this method will issue a warning when
+    // the type of a class defaults to something other than top. That's fine
+    // for this checker, because we don't expect user-written classes to take
+    // on values other than @NonConstant.
+  }
+
+  /**
+   * The standard override checker is replaced so that no override.param.invalid errors are issued
+   * when overriding methods defined in bytecode (which therefore use a different, incompatible
+   * defaulting scheme).
+   */
+  @Override
+  protected OverrideChecker createOverrideChecker(
+      Tree overriderTree,
+      AnnotatedExecutableType overrider,
+      AnnotatedTypeMirror overridingType,
+      AnnotatedTypeMirror overridingReturnType,
+      AnnotatedExecutableType overridden,
+      AnnotatedTypeMirror.AnnotatedDeclaredType overriddenType,
+      AnnotatedTypeMirror overriddenReturnType) {
+    return new NoLiteralOverrideChecker(
+        overriderTree,
+        overrider,
+        overridingType,
+        overridingReturnType,
+        overridden,
+        overriddenType,
+        overriddenReturnType);
+  }
+
+  private class NoLiteralOverrideChecker extends OverrideChecker {
+    /**
+     * Create an OverrideChecker.
+     *
+     * <p>Notice that the return types are passed in separately. This is to support some types of
+     * method references where the overrider's return type is not the appropriate type to check.
+     *
+     * @param overriderTree the AST node of the overriding method or method reference
+     * @param overrider the type of the overriding method
+     * @param overridingType the type enclosing the overrider method, usually an
+     *     AnnotatedDeclaredType; for Method References may be something else
+     * @param overridingReturnType the return type of the overriding method
+     * @param overridden the type of the overridden method
+     * @param overriddenType the declared type enclosing the overridden method
+     * @param overriddenReturnType the return type of the overridden method
+     */
+    public NoLiteralOverrideChecker(
+        Tree overriderTree,
+        AnnotatedExecutableType overrider,
+        AnnotatedTypeMirror overridingType,
+        AnnotatedTypeMirror overridingReturnType,
+        AnnotatedExecutableType overridden,
+        AnnotatedTypeMirror.AnnotatedDeclaredType overriddenType,
+        AnnotatedTypeMirror overriddenReturnType) {
+      super(
+          overriderTree,
+          overrider,
+          overridingType,
+          overridingReturnType,
+          overridden,
+          overriddenType,
+          overriddenReturnType);
+    }
+
+    @Override
+    public boolean checkOverride() {
```

Comment:
Please say what this is doing (that's different than the superclass).  That might be a code comment at the beginning of the method rather than a Javadoc comment -- I don't know what is appropriate.

Reply:
The documentation appears in the Javadoc comment for `NoLiteralVisitor#createOverrideChecker()`, which I thought (and still think) is the most appropriate place for it. I will add a summary comment here, however.